crates/transaction-pool/src/traits.rs

```diff
@@ -176,6 +176,42 @@ pub trait TransactionPool: Clone + Debug + Send + Sync {
         transactions: Vec<Self::Transaction>,
     ) -> impl Future<Output = Vec<PoolResult<TxHash>>> + Send;
 
+    /// Submit a consensus transaction directly to the pool
+    fn add_consensus_transaction(
+        &self,
+        tx: Recovered<<Self::Transaction as PoolTransaction>::Consensus>,
+        origin: TransactionOrigin,
+    ) -> impl Future<Output = PoolResult<TxHash>> + Send {
+        async move {
+            let tx_hash = *tx.tx_hash();
+
```

Comment:
this is a bit unfortunate,

what we could do is change the error type and require that this at least the returns the tx hash, e.g.:

https://github.com/paradigmxyz/reth/blob/1e208710432fcb0d3a4b367a3c3977661305a1d1/crates/transaction-pool/src/traits.rs#L1248-L1249

but we can do this in a separate step